src/main/groovy/wooga/gradle/unity/internal/DefaultUnityPluginExtension.groovy

```diff
@@ -17,531 +17,29 @@
 import wooga.gradle.unity.UnityPluginExtension
-import wooga.gradle.unity.batchMode.*
-import wooga.gradle.unity.batchMode.internal.DefaultActivationActionFactory
-import wooga.gradle.unity.batchMode.internal.DefaultBatchModeActionFactory
-import wooga.gradle.unity.tasks.SetAPICompatibilityLevel
-import wooga.gradle.unity.utils.internal.UnityHub
+import wooga.gradle.unity.models.APICompatibilityLevel
+import wooga.gradle.unity.models.BuildTarget
+import wooga.gradle.unity.models.UnityAuthentication
+import wooga.gradle.unity.utils.UnityHub
 
-import java.util.concurrent.Callable
+class DefaultUnityPluginExtension implements UnityPluginExtension {
 
-import static org.gradle.util.ConfigureUtil.configureUsing
-
-class DefaultUnityPluginExtension implements UnityPluginExtension, UnityPluginActionExtension {
-
-    static File defaultUnityLocation() {
-        File unityPath = null
-        String osName = System.getProperty("os.name").toLowerCase()
-        String osArch = System.getProperty("os.arch").toLowerCase()
-
-        if (osName.contains("windows")) {
-            if (osArch.contains("64")) {
-                unityPath = UnityPluginConsts.UNITY_PATH_WIN
-            } else {
-                unityPath = UnityPluginConsts.UNITY_PATH_WIN_32
-            }
-        } else if (osName.contains("linux")) {
-            unityPath = UnityPluginConsts.UNITY_PATH_LINUX
-        } else if (osName.contains("mac os x")) {
-            unityPath = UnityPluginConsts.UNITY_PATH_MAC_OS
-        }
-        unityPath
-    }
-
-    private Boolean autoReturnLicense
-    private Boolean autoActivateUnity
-    private Boolean batchModeForPlayModeTest
-    private Boolean batchModeForEditModeTest
-
-    private final Instantiator instantiator
-    protected final FileResolver fileResolver
     private final Project project
-    private final UnityAuthentication authentication
 
-    Factory<BatchModeAction> batchModeActionFactory
-    Factory<ActivationAction> activationActionFactory
-
-    private Factory<File> reportsDir
-    private Factory<File> assetsDir
-    private Factory<File> pluginsDir
-    private Factory<File> customUnityPath
-    private Object defaultBuildTarget
-    private final List<Object> testBuildTargets = new ArrayList<Object>()
-    private Boolean redirectStdOut = false
-    private String logCategory
+    DefaultUnityPluginExtension(Project project) {
+        this.project = project
+    }
 
-    static File getUnityPathFromEnv(Map<String, ?> properties, Map<String, String> env) {
-        String unityPath = properties[UnityPluginConsts.UNITY_PATH_OPTION] ?: env[UnityPluginConsts.UNITY_PATH_ENV_VAR]
+    File getDefaultUnityPath(Map<String, ?> properties, Map<String, String> env = null) {
+        String unityPath = UnityPluginConventions.unityPath.getValue(properties, env)
         if (unityPath) {
             return new File(unityPath)
         } else {
             return null
         }
     }
```

Comment:
That should no longer be needed implementated like this.

Reply:
Aye, I left it since an older test was using it, but I should update that test instead.